app/views/media_objects/_add_to_playlist.html.erb

```diff
@@ -0,0 +1,258 @@
+            <%= label_tag nil, t("playlist.visibility") %>
+            <label>
+              <%= f.radio_button(:visibility, Playlist::PUBLIC) %>
+              <%= human_friendly_visibility Playlist::PUBLIC %>
+            </label>
+            <label class="ml-2">
+              <%= f.radio_button(:visibility, Playlist::PRIVATE) %>
+              <%= human_friendly_visibility Playlist::PRIVATE %>
+            </label>
+          </div>
+          <div class="form-group text-right">
+            <%= f.submit id: 'new_playlist_submit', class: 'btn btn-primary', value: t("playlist.create.action") %>
+          </div>
+        </div><!-- /.modal-body -->
+      <% end # form_for playlist_form%>
+    </div><!-- /.modal-content -->
+  </div><!-- /.modal-dialog -->
+</div><!-- /.modal -->
+
+
+<% content_for :page_scripts do %>
+  <script>
+    $(document).ready(function() {
+      let mediaObjectId, streamId, mediaObjectTitle;
+      let listenersAdded = false;
+      // Enable add to playlist button after derivative is loaded
+      let timeCheck = setInterval(initAddToPlaylist, 500);
+      function initAddToPlaylist() {
+        let player = document.getElementById('iiif-media-player');
+        let scopes = [];
+        if(player) {
+          addPlayerEventListeners(player.player, () => {
+            clearInterval(timeCheck);
+          });
+        }
+      }
+
+      function addPlayerEventListeners(player) {
+        player.on('loadedmetadata', () => {
+          enableAddToPlaylist();
+        });
+        player.on('timeupdate', () => {
+          if(getActiveItem() != undefined) {
+            let { label, times, streamId } = getActiveItem();
+            $('#playlist_item_start').val(createTimestamp(times.begin, true));
+            $('#playlist_item_end').val(createTimestamp(times.end, true));
+            $('#playlist_item_title').val(`${mediaObjectTitle} - ${label}`);
+            $('#current-track-name').text(`${mediaObjectTitle} - ${label}`);
+          }
+        });
+        player.on('dispose', () => {
+          $('#addToPlaylistPanel').collapse('hide');
+          resetAddToPlaylistForm();
+          addToPlaylistBtn.disabled = true;
+        });
+      }
+
+      function enableAddToPlaylist() {
+        let addToPlaylistBtn = document.getElementById('addToPlaylistBtn');
+        if(addToPlaylistBtn) {
+          addToPlaylistBtn.disabled = false;
+        }
+        if(!listenersAdded) {
+          addListeners();
+        }
+      }
+
+      function addListeners() {
+        $('#addToPlaylistPanel').on('show.bs.collapse', function (e) {
+          // Hide share resource panel on add to playlist panel collapse
+          $('#shareResourcePanel').collapse('hide');
+          // Hide add to playlist alert on panel collapse
+          $('#add_to_playlist_alert').slideUp(100);
+          // Show add to playlist form on collapse and reset initially
+          $('#add_to_playlist_form_group').slideDown();
```

Comment:
I was confused at first by these comments.  What do you think about changing the wording?
```suggestion
          // Hide share resource panel on add to playlist panel show
          $('#shareResourcePanel').collapse('hide');
          // Hide add to playlist alert on panel show
          $('#add_to_playlist_alert').slideUp(100);
          // Show add to playlist form on show and reset initially
          $('#add_to_playlist_form_group').slideDown();
```